### PR TITLE
[#1431 , #1445 , #1594] feature: Add get_value method to MongoDB and Redis FSM storages

### LINKED ISSUE #1431: Add function get_value to FSMContext
### aiogram version

3.x

### Problem

In a situation where you only need to take one value in FSMContext handler, you need to write 2 lines of code to take value

### Possible solution

Add a get_value function for FSMContext that takes value by key

### Alternatives

_No response_

### Code example

```python3
# before
data = await state.get_data()
name = data["name"]

# after
name = await state.get_value("name")
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Vadim-Khristenko**
# Description

Added `get_value` method to MongoDB and Redis FSM storages to retrieve individual values from stored user data. This completes the implementation across all storage backends.

For MongoDB, the implementation uses projection queries to optimize performance when retrieving single values from large data objects.

Fixes #1431  
Supersedes inactive PR #1445 (no updates since June 18, 2024)  
Complements PR #1594 (which added the method to MemoryStorage and Scenes)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- [x] Comprehensive MongoDB-specific tests including projection optimization verification
- [x] Basic Redis functionality tests (samples available at [this gist](https://gist.github.com/Vadim-Khristenko/2b17d6c72a225347b58391b0ee33501e)) - can be added to the main test suite if needed
- [x] Integration with existing storage test cases
- [x] Default value handling and edge cases

**Test Configuration**:
* Operating System: Ubuntu 20.04.6 LTS (GitHub Codespace)
* Python version: 3.12.9 (Conda)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: aiogram/fsm/storage/redis.py [hunk 1]
```diff
@@ -127,6 +127,18 @@ async def get_data(
...
         return cast(Dict[str, Any], self.json_loads(value))
 
+    @overload
+    async def get_value(self, storage_key: StorageKey, dict_key: str) -> Optional[Any]: ...
+
+    @overload
+    async def get_value(self, storage_key: StorageKey, dict_key: str, default: Any) -> Any: ...
+
+    async def get_value(
+        self, storage_key: StorageKey, dict_key: str, default: Optional[Any] = None
```
Comment (JrooTJunior) [inline aiogram/fsm/storage/redis.py:137]: You don't need to override this method in the Redis Storage because the base implementation is the same as you have added.

File: aiogram/fsm/storage/mongo.py [hunk 2]
```diff
@@ -115,6 +115,23 @@ async def get_data(self, key: StorageKey) -> Dict[str, Any]:
...
         return cast(Dict[str, Any], document["data"])
 
+    @overload
+    async def get_value(self, storage_key: StorageKey, dict_key: str) -> Optional[Any]: ...
+
+    @overload
+    async def get_value(self, storage_key: StorageKey, dict_key: str, default: Any) -> Any: ...
+
+    async def get_value(
+        self, storage_key: StorageKey, dict_key: str, default: Optional[Any] = None
```
Comment (JrooTJunior) [inline aiogram/fsm/storage/mongo.py:125]: Actually what is the difference with the original implementation in the base method?

Is the same as get_data + get_value, so it is not needed to be overriden

**Review by JrooTJunior — CHANGES_REQUESTED** [review]
I don't see any important changes that add or improve anything, so why should these changes be accepted?

**Comment by Vadim-Khristenko** [comment]
I have decided to close this pull request due to its lack of necessity and relevance. I sincerely apologize for wasting your time, @JrooTJunior.
